app/src/main/java/org/mozilla/tv/firefox/search/SearchEngineProviderWrapper.kt

```diff
@@ -48,11 +48,15 @@ class SearchEngineProviderWrapper(private val replacements: Map<String, String>)
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     fun updateSearchEngines(searchEngines: SearchEngineList, replacements: Map<String, String>): SearchEngineList {
-        val defaultSearch = searchEngines.default
+        var defaultSearch = searchEngines.default
+
         @Suppress("NAME_SHADOWING") // Defensive copy & mutable
         val searchEngines = mutableListOf<SearchEngine>().apply { addAll(searchEngines.list) }
 
         replacements.forEach { (old, new) ->
+            if (defaultSearch?.identifier == old) {
+                defaultSearch = searchEngines[searchEngines.indexOfFirst { it.identifier == new } ]
```

Comment:
nit:
```kotlin
defaultSearch = searchEngines.firstOrNull { it.identifier == new }
```

And if we don't want to replace it with null:
```kotlin
defaultSearch = searchEngines.firstOrNull { it.identifier == new } ?: defaultSearch
```

Or just put it inside the `newIndex != -1` block.